.buildkite/scripts/serverless/create_deploy_tag/release_events.ts

```diff
@@ -0,0 +1,312 @@
+
+  buildkite.setMetadata('release_state', targetStateName);
+  buildkite.setMetadata('state_data', JSON.stringify(stateData));
+
+  updateWizardState(stateData);
+  updateWizardInstruction(targetStateName, stateData);
+
+  return targetStateName;
+}
+
+function updateWizardState(stateData: Record<string, 'ok' | 'nok' | 'pending' | undefined>) {
+  const wizardHeader = `<h3>:kibana: Kibana Serverless deployment wizard :mage:</h3>`;
+
+  const wizardSteps = Object.keys(states)
+    .filter((stateName) => states[stateName].display)
+    .map((stateName) => {
+      const stateInfo = states[stateName];
+      const stateStatus = stateData[stateName];
+      const stateEmoji = {
+        ok: ':white_check_mark:',
+        nok: ':x:',
+        pending: ':hourglass_flowing_sand:',
+        missing: ':white_circle:',
+      }[stateStatus || 'missing'];
+
+      if (stateStatus === 'pending') {
+        return `<div>[${stateEmoji}] ${stateInfo.name}<br />&nbsp; - ${stateInfo.description}</div>`;
+      } else {
+        return `<div>[${stateEmoji}] ${stateInfo.name}</div>`;
+      }
+    });
+
+  const wizardHtml = `<section>
+${wizardHeader}
+${wizardSteps.join('\n')}
+</section>`;
+
+  buildkite.setAnnotation(WIZARD_CTX_DEFAULT, 'info', wizardHtml);
+}
+
+function updateWizardInstruction(targetState: string, stateData: any) {
+  const { instructionStyle, instruction } = states[targetState];
+
+  if (instruction) {
+    buildkite.setAnnotation(
+      WIZARD_CTX_INSTRUCTION,
+      instructionStyle || 'info',
+      `<strong>${instruction}</strong>`
+    );
+  }
+}
+
+async function tryCall(fn: any, ...args: any[]) {
+  if (typeof fn === 'function') {
+    try {
+      const result = await fn(...args);
+      return result !== false;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  } else {
+    return true;
+  }
+}
+
+function sendSlackAnnouncement(selectedCommit: string | null, deployTag: string | null) {
+  const isDryRun = process.env.DRY_RUN?.match('(1|true)');
+  const textBlock = (...str: string[]) => ({ type: 'mrkdwn', text: str.join('\n') });
+  const buildShortname = `kibana-serverless-release #${process.env.BUILDKITE_BUILD_NUMBER}`;
+  sendSlackMessage({
+    blocks: [
+      {
+        type: 'section',
+        text: textBlock(
+          `:ship: Promotion of a new <https://github.com/elastic/kibana/commit/${selectedCommit}|commit> to QA has been initiated!`,
+          `:mag: The details of the candidate selection can be found here: <${process.env.BUILDKITE_BUILD_URL}|${buildShortname}>`,
+          `:test_tube: Once promotion is complete, please begin any required manual testing.`,
+          `*Remember:* Promotion to Staging is currently a manual process and will proceed once the build is signed off in QA.`,
+          `${isDryRun ? '*:white_check_mark:This is a dry run, no action will be taken.*' : ''}`
+        ),
+      },
+      {
+        type: 'section',
+        fields: [
```

Comment:
Could you add the commit date to the fields (in UTC if possible)

Reply:
added